### ISSUE TITLE
ceph-salt status/apply sanity check: all minion nodes should have clean "zypper ps -s"

### ISSUE THREAD

**Issue by smithfarm** [ceph/ceph-salt#341]
If "zypper ps -s" is not clean on all nodes, `ceph-salt apply` should not initiate the Salt Formula.

**Comment by LenzGr** [comment]
Sadly, the [Salt Zypper module](https://docs.saltstack.com/en/latest/ref/modules/all/salt.modules.zypperpkg.html) does not seem to support this natively, so this would have to be implemented differently.

**Comment by smithfarm** [comment]
> Sadly, the [Salt Zypper module](https://docs.saltstack.com/en/latest/ref/modules/all/salt.modules.zypperpkg.html) does not seem to support this natively, so this would have to be implemented differently.

Right. This will be implemented by running `zypper ps` and grepping the output for the string "No processes using deleted files found". (We are already using this in `ceph-salt reboot` and `ceph-salt update --reboot` to determine whether nodes need to be rebooted.)

**Comment by sebastian-philipp** [comment]
Working around issues of the Salt Zypper module seems strange to me.

**Comment by smithfarm** [comment]
> Working around issues of the Salt Zypper module seems strange to me.

Sounds like you haven't had the honor of working with this module before :-)

**Comment by sebastian-philipp** [comment]
> > Working around issues of the Salt Zypper module seems strange to me.
> 
> Sounds like you haven't had the honor of working with this module before :-)

haha, not yet :smile:
